--- dfu.c	2014-03-13 08:03:19.000000000 +0900
+++ ../../A/dfu.c	2015-10-31 06:57:42.795840300 +0900
@@ -36,13 +36,14 @@
 
 #ifndef HAVE_LIBUSB
 
-int dfu_open(struct dfu_dev *dfu, char *port_name) {
+struct dfu_dev * dfu_open(char *port_spec) {
   fprintf(stderr, "%s: Error: No USB support in this compile of avrdude\n",
     progname);
-  return -1;
+  return NULL;
 }
 
-int dfu_init(struct dfu_dev *dfu, unsigned short usb_pid) {
+int dfu_init(struct dfu_dev *dfu,
+  unsigned short vid, unsigned short pid) {
   return -1;
 }
 
