diff -urN libusb-1.0.20.orig/libusb/os/windows_usb.h libusb-1.0.20/libusb/os/windows_usb.h
--- libusb-1.0.20.orig/libusb/os/windows_usb.h	     2015-09-06 02:20:00.000000000 +0900
+++ libusb-1.0.20/libusb/os/windows_usb.h	     2015-10-31 12:55:34.000000000 +0900
@@ -48,7 +48,7 @@
 #endif
 
 #if defined(__CYGWIN__ )
-#define _stricmp stricmp
+#define _stricmp strcasecmp
 #define _snprintf snprintf
 #define _strdup strdup
 // _beginthreadex is MSVCRT => unavailable for cygwin. Fallback to using CreateThread

